doc/designs/disable-stale-users.md

```diff
@@ -0,0 +1,281 @@
+# Disable Stale Users
+
+**DESIGN STAGE**
+
+## Overview
+
+Some regulations call for accounts that haven't been used for a specific
+amount of time be automatically disabled in the identity management system.
+
+FreeIPA does not keep track of successful authentication events.
+The krbLastSuccessfulAuth attribute is not updated by default since
+https://pagure.io/freeipa/issue/5313 for reliability reasons.
+Therefore distinguishing between stale and active accounts cannot be done with
+existing logon data.
+
+
+## Possible approaches
+
+### Environments with mandatory, periodic password changes
+
+The problem is easy to solve in environments where user passwords must be
+changed regularly. A user account whose associated password is past its due
+change date can be considered stale and therefore be disabled by the system.
+In this case, a grace period should be provided between password expiration and
+the actual account locking event.
+Total writes to LDAP using this approach due to the disable stale users tooling
+are minimal (only nsAccountLock). However, password-change related writes are
+needed every 90 day (by default).
+Each password change updates the krbLastPwdChange, krbPasswordExpiration, and
+krbExtraData attributes.
```

Comment:
being pedantic, a password change also updates userPassword and krbPrincipalKey.

Reply:
Fixed, thanks!